### ISSUE TITLE
Safari Auto-Away

### ISSUE THREAD

**Issue by noezdev** [chatwoot/chatwoot#5076]
**Describe the bug**

When logged into Safari as an agent (Mac OS) and Chatwoot is in a tab that is not in the foreground, the status changes to Offline after a short time. You can prevent this by setting the auto_offline status of the agent to "false" via API (there is no option for this in the GUI?). However, this only shifts the problem, because now the status switches after a short time, although the agent is available.

If the agent has e.g. the iOS app open in parallel, the status changes constantly between online/away. This is very confusing for a website visitor.

This behavior is not reproducible with Firefox

**To Reproduce**

1. Login with Safari on MacOS and open a new tab so chatwoot is in the background
2. Go to the website with the widget and wait 5 minutes
3. The message that you are away will shown in the widget

**Environment**

Self hosted Linux

**Comment by pranavrajs** [comment]
@noezdev, A couple of points here.

1. The online status of agents is decided based on whether they have an active session in a browser or an app. This is done mostly by their active websocket connection. Since you are using the self-hosted installation, you might need to update the config to ensure that the server doesn't block long living connections. Please see more information here. https://www.chatwoot.com/docs/self-hosted/deployment/troubleshooting/#disconnected-from-chatwoot

2. The availability on the widget is designed based on the business hours. If you have not set the business hours, then only we would take the availability of agents on the platform. 

Let me know if it doesn't solve the problem.

**Comment by noezdev** [comment]
Hello,

I understand. However, I assume that since it only occurs with Safari, that Safari simply won't let a background tab send anything. The problem also occurs on app.chatwoot. The back and forth with the online status, as long as iOS is running in addition, will probably be due to the interval of iOS reporting "Hey, the agent is here", but the session in Safari acts as "away". Difficult to describe, since I also lack the background knowledge.

A simple solution would be to set both Auto-Away and Auto-Offline for agents via GUI. Auto-Offline can already be disabled via API.

The pure business hours are practical for large companies, but not for all companies, because you may want to offer the chat only if really enough capacity is available.

And as long as an agent proactively logs in as online in the browser, in this case Safari, he/she is really ready for chat sessions in our case. If the agent is still working in other tabs while no chats are coming in, the agent is still as ready as if it had the tab open. But for the customer, it makes it seem like the agent is currently absent and may decide not to chat due to this fact.

In another issue I already suggested that you should be able to hide the chat widget when all agents are offline. We currently solve this with a workaround (php script displays the widget if an available agent is found in the API, otherwise not).

**Comment by pranavrajs** [comment]
<img width="260" alt="Screenshot 2023-01-12 at 2 27 25 PM" src="https://user-images.githubusercontent.com/2246121/212194340-33af7b32-73fa-4a0a-b259-63301c934ed2.png">


Closing this as the toggle to disable auto-offline is shipped in the latest releases.
